src/components/Autocomplete/AutocompleteContext.tsx

```diff
@@ -0,0 +1,19 @@
+import { createContext } from "preact"
+import { useContext } from "preact/hooks"
+import { SearchAnalyticsOptions } from "@nosto/nosto-js/client"
+
+type AutocompleteContextValue = {
+  onSubmit: (query: string, options?: SearchAnalyticsOptions) => void
+}
+
+const AutocompleteContext = createContext<AutocompleteContextValue | undefined>(undefined)
+
+export function useAutocompleteContext() {
+  const context = useContext(AutocompleteContext)
+  if (!context) {
+    throw new Error("useAutocompleteContext must be used within an AutocompleteContextProvider")
+  }
+  return context
+}
+
+export default AutocompleteContext
```

Comment:
don't export the AutocompleteContext, but a dedicated AutocompleteContextProvider, similar to SidebarContext